src/main/java/io/pravega/connectors/hadoop/PravegaInputFormat.java

```diff
@@ -99,4 +123,43 @@ protected PravegaInputFormat(ClientFactory externalClientFactory) {
     public RecordReader<EventKey, V> createRecordReader(InputSplit inputSplit, TaskAttemptContext context) throws IOException, InterruptedException {
         return new PravegaInputRecordReader<V>();
     }
+
+    public static String fetchLatestPositionsJson(String uri, String scopeName, String streamName) throws IOException {
+        Preconditions.checkArgument(uri != null && !uri.isEmpty());
+        Preconditions.checkArgument(scopeName != null && !scopeName.isEmpty());
+        Preconditions.checkArgument(streamName != null && !streamName.isEmpty());
+
+        String pos = "";
+        try (ClientFactory clientFactory = ClientFactory.withScope(scopeName, URI.create(uri))) {
+            pos = fetchPositionsJson(clientFactory, scopeName, streamName);
+        }
+        return pos;
+    }
+
+    @VisibleForTesting
+    public static String fetchPositionsJson(ClientFactory clientFactory, String scopeName, String streamName) throws IOException {
+        BatchClient batchClient = clientFactory.createBatchClient();
+        StreamSegmentsIterator ssIter = batchClient.getSegments(new StreamImpl(scopeName, streamName), null, null);
```

Comment:
This can be replaced by `BatchClient#getStreamInfo` api ref: https://github.com/pravega/pravega/pull/2454

Reply:
as soon as that PR is closed